Core/GDCore/IDE/Events/EventsListUnfolder.h

```diff
@@ -19,4 +19,29 @@ class GD_CORE_API EventsListUnfolder {
       }
     }
   }
+
+  static void FoldAll(gd::EventsList& list) {
+    for (size_t i = 0; i < list.size(); ++i) {
+      gd::BaseEvent& event = list[i];
+      event.SetFolded(true);
+      if (event.CanHaveSubEvents() && event.GetSubEvents().size() > 0) {
+        FoldAll(event.GetSubEvents());
+      }
+    }
+  }
+
+  static void UnfoldToLevel(gd::EventsList& list,
+                            const int8_t maxLevel,
+                            const std::size_t currentLevel = 0) {
+    if (maxLevel >= 0 && currentLevel > maxLevel) return;
+
+    for (size_t i = 0; i < list.size(); ++i) {
+      gd::BaseEvent& event = list[i];
+      event.SetFolded(false);
+      if (event.CanHaveSubEvents() && event.GetSubEvents().size() > 0 &&
+          (maxLevel == -1 || currentLevel <= maxLevel)) {
```

Comment:
maybe I'd add a comment above the function, explaining that when `maxLevel == -1` it means, unfold until reaching max